[Bug]: Error on GitHub OAuth callback on local

### Is there an existing issue for the same bug?

- [x] I have checked the existing issues.

### Describe the bug and reproduction steps

`git pull upstream main`
`make build`

```

.react-router/types/src/routes/+types/oauth.github.callback.ts:8:29 - error TS2307: Cannot find module '../oauth.github.callback.js' or its corresponding type declarations.

8 type Module = typeof import("../oauth.github.callback.js")
                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 1 error in .react-router/types/src/routes/+types/oauth.github.callback.ts:8

make[1]: *** [build-frontend] Error 2
make: *** [build] Error 2
```

Some saas dependency got into the regular build?

### OpenHands Installation

Docker command in README

### OpenHands Version

_No response_

### Operating System

None

### Logs, Errors, Screenshots, and Additional Context

_No response_

--- a/frontend/package.json
+++ b/frontend/package.json
@@ -50,7 +50,7 @@
   "scripts": {
     "dev": "npm run make-i18n && cross-env VITE_MOCK_API=false react-router dev",
     "dev:mock": "npm run make-i18n && cross-env VITE_MOCK_API=true react-router dev",
-    "build": "npm run make-i18n && tsc && react-router build",
+    "build": "npm run make-i18n && npm run typecheck && react-router build",
     "start": "npx sirv-cli build/ --single",
     "test": "vitest run",
     "test:e2e": "playwright test",

